jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ServletContextResponse.java

```diff
@@ -337,12 +338,12 @@ public String getCharacterEncoding(boolean setContentType)
         
         // Try charset assumed from content type (assumed charsets are not added to content type header).
         MimeTypes mimeTypes = getRequest().getContext().getMimeTypes();
-        encoding = mimeTypes.getCharsetAssumedFromContentType(_contentType);
-        if (encoding != null)
+        encoding = mimeTypes.getAssumedCharsetName(_contentType);
+        if (encoding != null && !encoding.isEmpty())
```

Comment:
If the assumed is the empty string, then it has no charset, so we should not trickle down and run the inferred logic, no? 
As we already established that there is no charset for that content-type, so we cannot/should not infer one.